### ISSUE TITLE
Multitouch in touchpad

### ISSUE THREAD

**Issue by paulhandy** [nuclearsandwich/surface3-archlinux#17]
Under the current recommendations for getting the typecover3 to work, two-finger scroll, along with any other multitouch gestures does not work. There has been a mention of it working on the live boot of manjaro, but I have not confirmed that observation.
Has anyone made any progress on these grounds?

**Comment by nuclearsandwich** [comment]
I don't really use the TypeCover and because of the missing battery stuff and my lack of adventurous spirit to compile kernels for fun (being a grownup is boring) I haven't been using the 3.19 kernel much.

I want to get user namespaces for LXC so I may have to roll a kernel anyway. If I do, I'll see if this works at all.

**Comment by altercation** [comment]
I don't use a type cover but saw this posted to the surfacelinux subreddit just now (possible working patch for multitouch on type cover)

https://gist.github.com/felipeota/afb5f510f5b315f8bed8

http://www.reddit.com/r/SurfaceLinux/comments/2zz2wl/patch_multitouch_on_typecover_3/

**Comment by matthewwardrop** [comment]
Just a reminder that you can use linux-surfacepro3 from Arch's AUR repositories, which will give you full touchpad support (based on slightly modified versions of the patches linked above, in order to make them work on the latest kernels).

**Comment by nuclearsandwich** [comment]
@matthewwardrop thanks for the update. I really need to pull some of the info people are collecting into the repository itself. If anyone wants to kick off a PR, that would be rad too.

**Comment by pacomaro** [comment]
I haven't been able to get the touchpad working even with the AUR kernel package.
Well...I'm using Gnome. It doesn't work in the regular login, but it does work with the wayland login. However, loging in with wayland disables the pen...

The touchpad options show up in the settings but the best I can do is make a regular click.
No secondary nor taps are recognized, much less scrolling.

**Comment by matthewwardrop** [comment]
Have you tried installing xf86-input-libinput?

**Comment by pacomaro** [comment]
well...that was dumb...
Thanks!

**Comment by matthewwardrop** [comment]
No, not dumb... I should probably add that to the README file on my package. Thanks for pointing it out!

**Comment by pacomaro** [comment]
I've noticed that after installing xf86-input-libinput the Touchscreen starts behaving a little buggy. (kinda like it does on wayland)
The major issues I've noticed occurring are:
1.The pointer doesn't always follow my touch.(get's stuck)
2.It clicks way off target.

**Comment by matthewwardrop** [comment]
Interesting... this isn't something I've noted to be the case in my setup
:/. Do you have any config files in xorg.conf.d?

**Comment by pacomaro** [comment]
On /etc/X11 nothing that refers a Touchscreen
On /usr/share/X11 I do have a 90-libinput.conf that refers to a Touchscreeen.

Thinking it may be in some kind of conflict I opted to comment out that part and so far it seems to be working fine. I'll leave it like this for some time and see how it goes.

**Comment by pacomaro** [comment]
So...just a heads up:

Updated the repo kernel to 4.3.3-2 which was released on Dec 31.
Booted to it this morning. Everything works fine except for the touchpad which is acting as a pointer (no scroll) even on the login screen.
This leads me to think it's something related to libinput, since I've been told the GDM login runs on libinput.

**Comment by nuclearsandwich** [comment]
In theory libinput will be more amenable to custom configuration than even the synaptics driver but it will require a lot of re-learning. I've sadly re-flashed my Surface back to Windows (I could never stomach sacrificing HD space to dual boot) so I can't test anything at the moment.

**Comment by pacomaro** [comment]
Yeah well...for me it will have to be learning. period. I'm not that knowing about linux (though I've trying it a lot over the years) recently I decided to take a leap and I've been working exclusively on linux for about 1 month. I chose Arch because it's rolling, because I could get it to run properly on the SP3, and because I want to learn. Right now I'm only keeping windows for warranty reasons.
But there's a lot to learn to get this thing just right!
